recipes/mender-client/debian-master/rules

```diff
@@ -1,42 +1,24 @@
 #!/usr/bin/make -f
+#
+#DH_VERBOSE=1
+# see FEATURE AREAS in dpkg-buildflags(1)
+#export DEB_BUILD_MAINT_OPTIONS = hardening=+all
+
+# see ENVIRONMENT in dpkg-buildflags(1)
+# package maintainers to append CFLAGS
+#export DEB_CFLAGS_MAINT_APPEND  = -Wall -pedantic
+# package maintainers to append LDFLAGS
+#export DEB_LDFLAGS_MAINT_APPEND = -Wl,--as-needed
 
 %:
 	dh $@
 
-override_dh_auto_build:
-	make CGO_CFLAGS="${CGO_CFLAGS}" CGO_LDFLAGS="${CGO_LDFLAGS}" CC="${CC}" \
-	GOOS=linux CGO_ENABLED=1 build
-
-install_targets_exclude=install-systemd install-modules-gen
-install_targets_all=$(shell grep -E '^install\-.*:' Makefile | cut -d: -f1)
-install_targets=$(filter-out $(install_targets_exclude),$(install_targets_all))
-override_dh_auto_install:
-	DESTDIR=debian/mender-client make $(install_targets)
-
-	# Install scripts/version file
-	install -d -m 755 debian/mender-client/etc/mender/scripts/
-	echo -n "3" > debian/mender-client/etc/mender/scripts/version
-
-	# MEN-6060: Make identity and inventory scripts into conf files so that
-	# they can be modified without risking that they get lost on an
-	# upgrade. To follow Debian policy, we must then put them in /etc and
-	# symlink to them, we can not simply mark them as conffiles:
-	# https://www.debian.org/doc/debian-policy/ch-files.html#location
-	install -d -m 755 debian/mender-client/etc/mender/identity
-	mv debian/mender-client/usr/share/mender/identity/* debian/mender-client/etc/mender/identity/
-	for file in debian/mender-client/etc/mender/identity/*; do \
-		ln -sf /etc/mender/identity/"$$(basename "$$file")" debian/mender-client/usr/share/mender/identity/; \
-	done
-
-	install -d -m 755 debian/mender-client/etc/mender/inventory
-	mv debian/mender-client/usr/share/mender/inventory/* debian/mender-client/etc/mender/inventory/
-	for file in debian/mender-client/etc/mender/inventory/*; do \
-		ln -sf /etc/mender/inventory/"$$(basename "$$file")" debian/mender-client/usr/share/mender/inventory/; \
-	done
+override_dh_auto_configure:
+	dh_auto_configure -- -D CMAKE_BUILD_TYPE=Release -D BOOST_ALL_DYN_LINK=ON
 
 override_dh_auto_test:
 	true
 
-override_dh_strip:
-	${CROSS_COMPILE}-strip --remove-section=.comment --remove-section=.note \
-	debian/mender-client/usr/bin/mender
+# This is required as we cheat when we install the debian packages for raspbian
+override_dh_shlibdeps:
+	dh_shlibdeps --dpkg-shlibdeps-params=--ignore-missing-info
```

Comment:
This, or we can install all the files from the debian packages, when we install them manually

Reply:
This is incorrect but I don't know how to solve it.

So previously we had for debian/armhf harcoded the dependencies ([here](https://github.com/mendersoftware/mender-dist-packages/pull/333/files#diff-2ae0f2ab603d72f1d5c372cd757066bbfb5399baa282f80e85172ebcc6abeabaL11-L13)) while for the rest we used the  `shlibs:Depends` magic. Now however we have set up `shlibs:Depends` for all and then changing here the target to make it "work" on debian/armhf.

On the high level, we should only make exceptions for `debian/armhf` and leave the other cases to build with the `shlibs:Depends`, but I don't have a good suggestion to make this work in the context of this PR.

Reply:
It's only ignoring the info files, not all dependencies, so it's not super critical. I suspect it's because we only install the library files themselves in our unpackaging, but don't have the time to find out right now.

For now i think this is Ok though

Reply:
Fixed locally by installing all the files during container build